app/src/main/java/com/example/pygmyhippo/user/QRFragment.java

```diff
@@ -40,15 +53,55 @@ public View onCreateView(@NonNull LayoutInflater inflater,
         return root;
     }
 
-    // qr code scanner to event button
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
-        Button scanQRButton = view.findViewById(R.id.u_scanQRButton);
-        scanQRButton.setOnClickListener(view1 -> {
-            Navigation.findNavController(view1).navigate(R.id.action_scanQRcodeFragment_to_eventFragment);
-        });
+        barcodeView = view.findViewById(R.id.u_barcodeScanner);
+        if (ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.CAMERA)
+                == PackageManager.PERMISSION_GRANTED) {
+            startSingleScan();
+        } else {
+            // request for camera permission
+            // TODO: with database, can check past permissions on profile
+            // maybe set initial to null, check if null, then set, can change in profile
+            ActivityCompat.requestPermissions(requireActivity(),
+                    new String[]{Manifest.permission.CAMERA},
+                    CAMERA_PERMISSION_REQUEST_CODE);
+        }
+    }
+
+    private void startSingleScan() {
+        barcodeView.decodeSingle(callback);
+    }
+
+    private final BarcodeCallback callback = result -> {
+        if (result != null && result.getText() != null) {
+            String eventID = result.getText();
+            Toast testShowBarcode = Toast.makeText(getActivity(), eventID, Toast.LENGTH_LONG);
+            testShowBarcode.show();
+
+            // TODO: if evenID doesnt correpond to anything in database, restart using startSingleScan()
+
+            Navigation.findNavController(requireView()).navigate(R.id.action_scanQRcodeFragment_to_eventFragment);
+        }
+    };
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (barcodeView != null && ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.CAMERA)
+                == PackageManager.PERMISSION_GRANTED) {
+            barcodeView.resume();
+        }
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        if (barcodeView != null) {
+            barcodeView.pause();
+        }
```

Comment:
honestly, not too sure myself, but need the overrides for it to work. i'm pretty sure it's because android studio has a resume and pause function already for when the activity is interacting/paused interacting with the user, and we need to override it because we want it to do something specific, specifically when scanning the qr code, the barcodeview itself continues or pauses